### ISSUE TITLE
`no-new-mixins`: `getSourceModuleName` should only be called on a `CallExpression`, `MemberExpression` or `Identifier`

### ISSUE THREAD

**Issue by buschtoens** [ember-cli/eslint-plugin-ember#905]
```ts
AssertionError [ERR_ASSERTION]: `getSourceModuleName` should only be called on a `CallExpression`, `MemberExpression` or `Identifier`
    at getSourceModuleName (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:46:5)
    at getSourceModuleName (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:42:12)
    at /Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:31:49
    at Array.some (<anonymous>)
    at /Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:30:36
    at Array.find (<anonymous>)
    at Object.getSourceModuleNameForIdentifier (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:29:6)
    at isEmberCoreModule (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/ember.js:213:46)
    at Object.isEmberMixin (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/ember.js:259:10)
    at ClassDeclaration (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/rules/no-new-mixins.js:36:19)
```

The above error appears to be triggered by code like:

```ts
test('basic functionality', function(this: TestContext, assert) {
  class TestObject extends this.ContainerObject {
    // ...
  }
});
```

**Comment by buschtoens** [comment]
Same for `no-attrs-in-components`.

```
AssertionError [ERR_ASSERTION]: `getSourceModuleName` should only be called on a `CallExpression`, `MemberExpression` or `Identifier`
    at getSourceModuleName (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:46:5)
    at getSourceModuleName (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:42:12)
    at /Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:31:49
    at Array.some (<anonymous>)
    at /Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:30:36
    at Array.find (<anonymous>)
    at Object.getSourceModuleNameForIdentifier (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:29:6)
    at isEmberCoreModule (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/ember.js:213:46)
    at isEmberComponent (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/ember.js:246:10)
    at ClassDeclaration (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/rules/no-attrs-in-components.js:34:13)
```

**Comment by buschtoens** [comment]
`no-get` is affected as well.

```
AssertionError [ERR_ASSERTION]: `getSourceModuleName` should only be called on a `CallExpression`, `MemberExpression` or `Identifier`
    at getSourceModuleName (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:46:5)
    at getSourceModuleName (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:42:12)
    at /Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:31:49
    at Array.some (<anonymous>)
    at /Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:30:36
    at Array.find (<anonymous>)
    at Object.getSourceModuleNameForIdentifier (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/import.js:29:6)
    at isEmberCoreModule (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/ember.js:213:46)
    at isEmberArrayProxy (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/ember.js:271:10)
    at Object.isEmberProxy (/Users/jan/open-source/decorators/node_modules/eslint-plugin-ember/lib/utils/ember.js:288:10)
```

I guess the issue is that `isEmberCoreModule` incorrectly passes the node through to `getSourceModuleNameForIdentifier`?

**Comment by rwjblue** [comment]
@buschtoens - Can you check what node type _is_ being passed (also we should update that assertion to tell you what was passed in addition to what is allowed, the message as it stands is a bit hard to debug)?

**Comment by rwjblue** [comment]
In general, `getSourceModuleNameForIdentifier` is only going to call `getSourceModuleName` on an import specifier, but when the `node.superClass` is not itself an `Identifier` (in this case its a member expression) there is an issue.

https://astexplorer.net/#/gist/b736a0d02d00957083b4a0118403fd7d/38043d9bc98eba1ab831d7ca16ddf57435504583

`getSourceModuleNameForIdentifier` is passed the `node.superClass` (which _normally_ is an `Identifier`, but in this case is a `MemberExpression`):

https://github.com/ember-cli/eslint-plugin-ember/blob/82acc5bac03c014c5828fb8dbf06dea6919e43f7/lib/utils/ember.js#L213-L216

Then `getSourceModuleNameForIdentifier` calls `getSourceModuleName(node)` with that `MemberExpression` and causes the error you are seeing:

https://github.com/ember-cli/eslint-plugin-ember/blob/82acc5bac03c014c5828fb8dbf06dea6919e43f7/lib/utils/import.js#L30-L32

Ultimately, I think we should update `isEmberCoreModule` to `return false` when we spot a non `Identifier` as `node.superClass`.

**Comment by bmish** [comment]
Opened fix in #906.
